backend/api/Controllers/RobotController.cs

```diff
@@ -732,10 +736,14 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
 
             await _missionRunService.Create(missionRun);
 
-            robot.Status = RobotStatus.Busy;
-            robot.CurrentMissionId = missionRun.Id;
-            robot.CurrentArea = area;
-            await _robotService.Update(robot);
+            try
+            {
+                await _robotService.UpdateRobotStatus(robot.Id, RobotStatus.Busy);
+                await _robotService.UpdateCurrentMissionId(robot.Id, missionRun.Id);
+                await _robotService.UpdateCurrentArea(robot.Id, area);
```

Comment:
Is it safer to do in 3 requests instead of 1?

Reply:
I think it would be better to do in one as we have additional database reads and writes although I don't think the difference is large as we have the semaphore now. I didn't want to make too many combination functions of type `UpdateRobotStatusAndCurrentMissionId...`.

For now will leave it like this and then we can consider improving it.